src/app/dim-api/reducer.ts

```diff
@@ -199,6 +199,19 @@ export const dimApi = (
 
     case getType(actions.profileLoaded): {
       const { profileResponse, account } = action.payload;
+
+      // If the user was using "element" sort, replace it with "elementWeapon" and "elementArmor" sorts.
+      const sortOrder = profileResponse.settings?.itemSortOrderCustom || [];
+      const reversals = profileResponse.settings?.itemSortReversals || [];
```

Comment:
This logic needs to be applied for the `profileLoadedFromIDB` action as well. You could make a function called `migrateSettings` or something.

Reply:
For clarification, do you mean everything (including replacing the occurrences of 'element') needs applied to `profileLoadedFromIDB` or just the two lines highlighted? I don't quite understand what the `profileLoadedFromIDB` action does different from `profileLoaded` so that is probably part of the confusion.

When you say I could make a new function is that to reduce repetitive code? So something like this...

```ts
case getType(actions.profileLoadedFromIDB): {
  // ...
  const [sortOrder, reversals] = migrateSettings(action.payload);
  // ...
}
case getType(actions.profileLoaded): {
  // ...
  const [sortOrder, reversals] = migrateSettings(profileResponse);
  // ...
}

function migrateSettings(payload) {
  // The logic/replacements
  return [sortOrder, reversals];
}
```

Reply:
`migrateSettings` should probably have the signature `(settings: Settings) => Settings`. The good thing is that in the meantime, we've already needed a function like this for a different purpose: 

https://github.com/DestinyItemManager/DIM/blob/d436fb07d95b47bb7c90d27ac25ebeb58f2a7339/src/app/dim-api/reducer.ts#L385-L403

So it would probably be best to rename this function to `migrateSettings`, adjust the comment, and add your logic there.